access/pagerduty/bot.go

```diff
@@ -209,46 +273,162 @@ func (b *Bot) CreateIncident(ctx context.Context, reqID string, reqData RequestD
+	if resp.IsError() {
+		return responseError(resp)
+	}
+
+	// TODO: PUT /incidents/{id}
+	var incidentsBody struct {
+		Incidents []pd.ManageIncidentsOptions `json:"incidents"`
+	}
+	incidentsBody.Incidents = []pd.ManageIncidentsOptions{
 		pd.ManageIncidentsOptions{
-			ID:     pdData.ID,
+			ID:     incidentID,
 			Type:   "incident_reference",
 			Status: "resolved",
 		},
-	})
-	return trace.Wrap(err)
+	}
+
+	resp, err = b.NewRequest(ctx).
+		SetHeader("From", b.from).
+		SetBody(&incidentsBody).
+		Put(b.APIURL("incidents"))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if resp.IsError() {
+		return responseError(resp)
+	}
+
+	return nil
 }
 
 func (b *Bot) GetUserInfo(ctx context.Context, userID string) (*pd.User, error) {
-	client := b.NewClient(ctx)
+	var userRes struct {
+		User pd.User `json:"user"`
+	}
+
+	resp, err := b.NewRequest(ctx).
+		SetResult(&userRes).
+		Get(b.APIURL("users", userID))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if resp.IsError() {
+		return nil, responseError(resp)
+	}
+
+	return &userRes.User, nil
+}
+
+func (b *Bot) GetUserByEmail(ctx context.Context, userEmail string) (*pd.User, error) {
+	usersOpts := pd.ListUsersOptions{
+		Query: userEmail,
+		APIListObject: pd.APIListObject{
+			Limit: pdListLimit,
+		},
+		// Total: true, // TODO:
+	}
 
-	user, err := client.GetUser(userID, pd.GetUserOptions{})
+	var usersRes pd.ListUsersResponse
+	usersOptsQuery, err := query.Values(usersOpts)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return user, nil
+
+	resp, err := b.NewRequest(ctx).
+		SetQueryParamsFromValues(usersOptsQuery).
+		SetResult(&usersRes).
+		Get(b.APIURL("users"))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if resp.IsError() {
+		return nil, responseError(resp)
+	}
+
+	for _, user := range usersRes.Users {
+		if user.Email == userEmail {
+			return &user, nil
+		}
+	}
+
+	return nil, trace.NotFound("failed to find pagerduty user by email %q", userEmail)
+}
+
+func (b *Bot) IsUserOnCall(ctx context.Context, userID string) (bool, error) {
+	var serviceRes struct {
+		Service pd.Service `json:"service"`
+	}
+	resp, err := b.NewRequest(ctx).
+		SetResult(&serviceRes).
+		Get(b.APIURL("services", b.serviceID))
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	if resp.IsError() {
+		return false, responseError(resp)
+	}
+
+	var onCallRes pd.ListOnCallsResponse
+	onCallOpts := pd.ListOnCallOptions{
+		UserIDs:             []string{userID},
+		EscalationPolicyIDs: []string{serviceRes.Service.EscalationPolicy.ID},
+	}
+	onCallOptsQuery, err := query.Values(onCallOpts)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	resp, err = b.NewRequest(ctx).
+		SetQueryParamsFromValues(onCallOptsQuery).
+		SetResult(&onCallRes).
+		Get(b.APIURL("oncalls"))
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	if resp.IsError() {
+		return false, responseError(resp)
+	}
+
+	return len(onCallRes.OnCalls) > 0, nil
```

Comment:
also confirm that the oncalls list contains a user with the right email
it's pagerduty API's job, just still good to double-check in case, e.g. `userID` was empty